regex2/challenges-11.test.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 1
+//
+// Write a function named validatePin that uses a regular expression pattern to validate a PIN.
+//
+// If the PIN is four numerical digits long, return true. Otherwise, return false.
+// ------------------------------------------------------------------------------------------------
+
+const validatePin = (pin) => {
+  // Solution code here...
+  return /^\d\d\d\d$/.test(pin)
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 2
+//
+// Write a function named findTagNames that iterates over an array of HTML strings
+// and uses a regular expression pattern to return the closing tags.
+//
+// For example, findTagNames(['<h1>Hello, world!</h1>', '<p>Welcome to my site</p>'])
+// returns ['/h1', '/p'], and findTagNames(['<div><h1>Hello, world!</h1></div>', '<p>Welcome to my site</p>'])
+// returns ['/h1', '/div', '/p'].
+// ------------------------------------------------------------------------------------------------
+
+const findTagNames = elements => {
+  // Solution code here...
+  return /\w*\/\w*+^/g.match(elements)

```

Comment:
The first `\w*` is actively unhelpful! That'll make sure you match any word characters that come before the slash. And you don't need both the star and the plus after the second `\w`, nor do you need the string to match at the end. And you'll need to iterate over the `elements` array to match against each individual string, not against the whole array.